src/main/java/io/nats/client/impl/NatsConnection.java

```diff
@@ -73,16 +64,16 @@
     static final byte LF = 0x0A;
     static final byte[] CRLF = { CR, LF };
 
-    static final String OP_CONNECT = "CONNECT";
-    static final String OP_INFO = "INFO";
-    static final String OP_SUB = "SUB";
-    static final String OP_PUB = "PUB";
-    static final String OP_UNSUB = "UNSUB";
-    static final String OP_MSG = "MSG";
-    static final String OP_PING = "PING";
-    static final String OP_PONG = "PONG";
-    static final String OP_OK = "+OK";
-    static final String OP_ERR = "-ERR";
+    static final ByteBuffer OP_CONNECT = ByteBuffer.wrap(new byte[] { 'C', 'O', 'N', 'N', 'E', 'C', 'T' });
+    static final ByteBuffer OP_INFO = ByteBuffer.wrap(new byte[] { 'I', 'N', 'F', 'O' });
+    static final ByteBuffer OP_SUB = ByteBuffer.wrap(new byte[] { 'S', 'U', 'B' });
+    static final ByteBuffer OP_PUB = ByteBuffer.wrap(new byte[] { 'P', 'U', 'B' });
+    static final ByteBuffer OP_UNSUB = ByteBuffer.wrap(new byte[] { 'U', 'N', 'S', 'U', 'B' });
+    static final ByteBuffer OP_MSG = ByteBuffer.wrap(new byte[] { 'M', 'S', 'G' });
+    static final ByteBuffer OP_PING = ByteBuffer.wrap(new byte[] { 'P', 'I', 'N', 'G' });
+    static final ByteBuffer OP_PONG = ByteBuffer.wrap(new byte[] { 'P', 'O', 'N', 'G' });
+    static final ByteBuffer OP_OK = ByteBuffer.wrap(new byte[] { '+', 'O', 'K' });
+    static final ByteBuffer OP_ERR = ByteBuffer.wrap(new byte[] { '-', 'E', 'R', 'R' });

```

Comment:
Since speed is not crucial at this initialization point it might be more readable to just do 
```
static final ByteBuffer OP_ERR = ByteBuffer.wrap("-ERR".getBytes());
```

Reply:
I did it this way mostly since it's pretty obvious from the variables names what they are, and since I'm wasn't sure there wouldn't be any unexpected performance issues due to allocation weirdness when calling `getBytes()`. I've seen instances where bytebuffers get sized weirdly when converting from strings sometimes since Java strings are essentially UTF-16 and not ASCII/UTF-8(which is what we need here), I figured this version was more obviously correct since it wouldn't be prone to those side effects as it entirely avoids Java UTF-16 formatted strings.

Reply:
Gotcha!

Reply:
Yeah, I've tried to fairly aggressively minimize the usage of strings wherever feasible, especially in the lower level serialization/de-serialization code as string conversions are relatively expensive operations that need to be aggressively minimized for performance. Strings are also just difficult to work with in general with lower level Java networking code since they make accurate buffer sizing difficult, hence why I'm trying to have everything immediately normalized to `ByteBuffer`'s internally as soon as feasible.